### DISCUSSION TITLE
Problems with private tap and GitHub Releases hosted bottles

### DISCUSSION THREAD

**Discussion by dan-zozo** [Homebrew/discussions#1534] [Tap maintenance and Homebrew development] [answered]
I have a private tap, and have managed to run Homebrew's GitHub Actions to produce a bottle for a private formula. Downloading the `*.tar.gz` via the web-page access to the release works well, but when attempting to install the formula via the CLI, I get a curl error:

```
curl: (22) The requested URL returned error: 404
```

Does Homebrew support the fetching and pouring of bottles that are hosted on _private_ GitHub repository taps?

I have tried adding token/creds to a `.netrc` file, but it seems like Homebrew/curl is not using this.

**Comment by carlocab** [comment, marked as answer]
Writing a [custom download strategy](https://docs.brew.sh/Formula-Cookbook#specifying-the-download-strategy-explicitly) for this might work.

See also https://github.com/Homebrew/brew/pull/11091 and https://github.com/Homebrew/brew/pull/3106 for related discussions.

**Reply by dan-zozo** [reply]
Thank you @carlocab :D this looks like an approach that could work, and some great discussions to read up on!

**Reply by dan-zozo** [reply]
Looking more at this, a custom download strategy _does_ work, but it wasn't easily obvious how to override the brew-provided curl download strategy in an extensible fashion that does not involve overriding private methods. There's always the option of developing a more fully-formed download strategy.

I did overlook that Homebrew provides a hook with the `HOMEBREW_CURLRC` environment variable that allows the use and injection of e.g. the `--netrc-optional` flag when executing curl. This gets me up and running with less fear around having to maintain a custom strategy.

**Reply by carlocab** [reply]
Ah, yes, that slipped my mind. Glad you found a nice solution.
